### DISCUSSION TITLE
gtk-single-instance=true and opening new window doesn't use up to date config

### DISCUSSION THREAD

**Discussion by barr-israel** [ghostty-org/ghostty#3225] [Q&A] [answered]
When setting `gtk-single-instance=true` and opening a new window via the DE, it will use a non-up to date config

reproduction:
1. set `gtk-single-instance=true` on some config X and reload config
2. change some config value to config Y
3. open new ghostty via DE
4. new ghostty will use config X instead of Y

Opening a new window via the original already open window will use config Y

Closing all ghostty windows(but not the background process left behind) and opening a new one, will use config X.

Expected behavior: all tabs and windows opened after the config has changed(especially after it has been reloaded), should use config Y

**Comment by barr-israel** [comment]
this issue somehow just disappeared after trying it again, after multiple times that it did happen, unsure what happened

**Comment by jcollie** [comment, marked as answer]
Ghostty only loads the config when starting up for the first time, and when explicitly told to do so via a keybind action. When `gtk-single-instance=true` and you launch Ghostty again, the new Ghostty process talks to the old Ghostty process and tells it to open up a new window, then the new Ghostty process exits. At no point in this will the old Ghostty re-read the configuration file. Ghostty does not watch the config file for changes either.

So you're seeing the expected behavior once you understand what's going on. If you make changes to the config file, you must explicitly tell Ghostty to reload the config file.

**Reply by rhodes-b** [reply]
it can be pointed out as well specific config options do not work by just reloading the config I believe this is one of them that requires a full restart
